src/midgard/util.cc

```diff
@@ -716,5 +721,35 @@ polygon_t to_boundary(const std::unordered_set<uint32_t>& region, const Tiles<Po
   return polygon;
 }
 
+constexpr char PADDING_ENCODED = '=';
+constexpr char ZERO_ENCODED = 'A';
+
+std::string encode64(const std::string& text) {
+  using namespace boost::archive::iterators;
+  using Base64Encode = base64_from_binary<transform_width<std::string::const_iterator, 6, 8>>;
+  // Encode and add padding to string per octet encoding described here:
+  // https://tools.ietf.org/html/rfc4648#section-4
+  std::string encoded(Base64Encode(text.begin()), Base64Encode(text.end()));
+  size_t num_pad_chars = (3 - text.size() % 3) % 3;
+  encoded.append(num_pad_chars, PADDING_ENCODED);
+  return encoded;
+}
+
+std::string decode64(const std::string& encoded) {
+  using namespace boost::archive::iterators;
+  using Base64Decode =
+      transform_width<binary_from_base64<remove_whitespace<std::string::const_iterator>>, 8, 6>;
+  // NOTE(mookerji): Ugh, for more details, see:
+  // https://stackoverflow.com/questions/10521581/base64-encode-using-boost-throw-exception
+  size_t num_pad_chars = (4 - encoded.size() % 4) % 4;
+  std::string padded = encoded;
```

Comment:
this copy is very unfortunate. i suppose that is the price you pay when you let boost do the decoding

Reply:
Yep!